code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -61,11 +62,21 @@ public boolean doWork(final Event event) {
 
     // Map containing the update event IDs (and corresponding requested scopes) for Edge events that
     // haven't yet received an Edge completion response.
-    private Map<String, List<DecisionScope>> updateRequestEventIdsInProgress = new HashMap<>();
+    // Updating the map type from Hashmap to ConcurrentHashMap to avoid the race condition to
+    // restrict it getting accessed by multiple threads
+    // Also declaring its reference as static and final because we have to make sure there is only
+    // one instance of the map for the singleton optimize extension class.
+    private static final Map<String, List<DecisionScope>> updateRequestEventIdsInProgress =
+            new ConcurrentHashMap<>();
 
     // a dictionary to accumulate propositions returned in various personalization:decisions events
     // for the same Edge personalization request.
-    private Map<DecisionScope, OptimizeProposition> propositionsInProgress = new HashMap<>();
+    // Updating the map type from Hashmap to ConcurrentHashMap to avoid the race condition to
+    // restrict it getting accessed by multiple threads.
+    // Also declaring its reference as static and final because we have to make sure there is only
+    // one instance of the map for the singleton optimize extension class.
+    private static final Map<DecisionScope, OptimizeProposition> propositionsInProgress =
```

Comment:
Same as above

Reply:
Updated the comments accordingly @praveek